packages/gateway/src/Shard.ts

```diff
@@ -117,20 +139,77 @@ export class DiscordenoShard {
     url.searchParams.set('v', this.gatewayConfig.version.toString())
     url.searchParams.set('encoding', 'json')
 
+    // Set the compress url param and initialize the decompression contexts
+    if (this.gatewayConfig.transportCompression) {
+      url.searchParams.set('compress', this.gatewayConfig.transportCompression)
+
+      if (this.gatewayConfig.transportCompression === TransportCompression.zlib) {
```

Comment:
(preference/stylistic, feel free to ignore) maybe switch case? idk I just think it make more sense in this case

Reply:
idk, i tend to avoid 2 cases switch statements. I generally think a few (2-4) if(s) are fine, after that a switch might be a better option